src/baldr/directededge.cc

```diff
@@ -515,6 +515,11 @@ void DirectedEdge::set_opp_local_idx(const uint32_t idx) {
   }
 }
 
+void DirectedEdge::set_hierarchy_roadclass(const uint64_t road_class, const bool reset) {
+  shortcut_ = road_class;
+  is_shortcut_ = !reset;
+}
```

Comment:
bit of a hack: during ferry & link evaluation, we flag `mjolnir::Edge` and in `mjolnir::DirectedEdgeBuilder` we remember the road class the edge should have wrt to hierarchies and store it in `shortcut_` which has 7 bits and isn't used until after the hierarchies (where we reset the shortcut attributes)